examples/preproc3-NVT_E2E.ipynb

```diff
@@ -0,0 +1,890 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_df.head(2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TIMESTAMP_DELTA = 1465876799998\n",
+    "def calculate_delta(col,gdf):\n",
+    "    delta = ((gdf['timestamp']+TIMESTAMP_DELTA).astype('datetime64[ms]') - col.astype('datetime64[ns]')).dt.days\n",
+    "    delta = delta * (delta >=0) * (delta<=10*365)\n",
+    "    return delta"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/ronayak/ronaya/NVTabular/nvtabular/io.py:748: UserWarning: part_mem_fraction is ignored for DataFrame input.\n",
+      "  warnings.warn(\"part_mem_fraction is ignored for DataFrame input.\")\n"
+     ]
+    }
+   ],
+   "source": [
+    "CATEGORICAL_COLUMNS =['display_id', 'ad_id', 'uuid', 'document_id','platform', 'geo_location', 'document_id_promo', 'campaign_id',\n",
+    "       'advertiser_id', 'source_id', 'publisher_id', 'publish_time','source_id_promo', 'publisher_id_promo', 'publish_time_promo']\n",
+    "CONTINUOUS_COLUMNS = ['timestamp']\n",
+    "\n",
+    "workflow = nvt.Workflow(\n",
+    "    cat_names=CATEGORICAL_COLUMNS,\n",
+    "    cont_names= CONTINUOUS_COLUMNS,\n",
+    "    label_name=['clicked'])\n",
+    "\n",
+    "workflow.add_preprocess([\n",
+    "    LambdaOp(\n",
+    "        op_name='day',\n",
+    "        f=lambda col, gdf: (col / 1000 / 60 / 60 / 24).astype(int),\n",
+    "        columns=['timestamp'], replace=False),\n",
+    "    LambdaOp(\n",
+    "        op_name='country',\n",
+    "        f=lambda col, gdf: col.str.slice(0,2),\n",
+    "        columns=['geo_location'], replace=False),\n",
+    "    LambdaOp(\n",
+    "        op_name='state',\n",
+    "        f=lambda col, gdf: col.str.slice(0,5),\n",
+    "        columns=['geo_location'],replace=False),\n",
+    "    LambdaOp(\n",
+    "        op_name='days_since_published',\n",
+    "        f=calculate_delta,\n",
+    "        columns=['publish_time','publish_time_promo'], replace=False),\n",
+    "    \n",
+    "    Categorify(columns=['uuid']),\n",
+    "    \n",
+    "    FillMedian(columns=['publish_time_days_since_published','publish_time_promo_days_since_published'])])\n",
+    "\n",
+    "workflow.add_feature([\n",
+    "    Dropna(['geo_location', 'platform']),\n",
+    "    JoinGroupby(columns=['ad_id', 'source_id', 'document_id_promo', 'publisher_id', 'advertiser_id', 'campaign_id'], \n",
+    "        cont_names=['clicked'],stats=['sum','count'])])\n",
+    "\n",
+    "workflow.finalize()\n",
+    "\n",
+    "GPU_MEMORY_FRAC = 0.15\n",
+    "train_dataset = nvt.Dataset(train_df, engine='parquet',  part_mem_fraction=GPU_MEMORY_FRAC)\n",
```

Comment:
if you're passing a dataframe, you don't need the engine / part_mem_fraction parameters 
```suggestion
    "train_dataset = nvt.Dataset(train_df)\n",
```